File: 프로그래머스/lv2/12980. 점프와 순간 이동/점프와 순간 이동.java
import java.util.*;

public class Solution {
    public int solution(int n) {
        int ans = 0;
        
        while(n > 0){
            // 점프를 수행하면 홀수번 칸에 도착 ex) 1, 3, 5
            if(n%2 == 1){
                ans++;
            }
            n = n/2;
        }
        
        return ans;
    }
}

File: 프로그래머스/2/12980. 점프와 순간 이동/점프와 순간 이동.py
def solution(n):
    ans = 0
    
    # 점프와 순간이동
    # 짝수일 때는 순간이동, 홀수일 때 점프
    while(n != 0):
        if n%2 == 0:
            n /= 2
        else:
            n -= 1
            ans += 1

    return ans